src/main/java/com/feedzai/commons/sql/abstraction/engine/DatabaseEngine.java

```diff
@@ -304,13 +304,16 @@
     List<Map<String, ResultColumn>> query(final Expression query) throws DatabaseEngineException;
 
     /**
-     * Executes the given query overriding the configured query timeout (see {@link PdbProperties#getSelectQueryTimeout()}).
+     * Executes the given query overriding the configured query timeout (see {@link PdbProperties#getSelectQueryTimeout()})
+     * if the underlying engine supports query timeouts.
      *
      * @param query The query to execute.
      * @param readTimeoutOverride The query timeout to use.
      * @throws DatabaseEngineException If something goes wrong executing the query.
      */
-    List<Map<String, ResultColumn>> query(final Expression query, final int readTimeoutOverride) throws DatabaseEngineException;
+    default List<Map<String, ResultColumn>> query(final Expression query, final int readTimeoutOverride) throws DatabaseEngineException {
+        return query(query);
```

Comment:
since `translate` is available as an interface method, you could instead use what the abstract engine is using:
```
        return query(translate(query), readTimeoutOverride);
```